components/tabs/src/tabs.udl

```diff
@@ -12,13 +21,16 @@ interface TabsStore {
 
     [Self=ByArc]
     void register_with_sync_manager();
+
+    [Throws=TabsError, Self=ByArc]
```

Comment:
I wonder if we should also allow the contructor to return this error? That's another breaking change, but it seems crazy that a sync function can return an error about a database being corrupt but the constructor will silence it?

Reply:
So I assumed because the `TabsStore::new` returns `Self` instead of `Result` that it doesn't return an error, which means it shouldn't (or wouldn't) throw a `TabsError`, let me know if there are otther considerations here. And while I do think it's off for the sync function to throw while the constructor doesn't, we do have a precedent for throwing on the sync functions we expose.

Reply:
oops, no, I thought we caught and ignored errors there, but apparently not!